int32 pipeline
---
int32 error